homeassistant/components/mqtt/client.py

```diff
@@ -688,6 +691,29 @@ def _process_client_subscriptions() -> list[tuple[int, int]]:
         if errors:
             _raise_on_errors(errors)
 
+    async def _async_perform_unsubscribes(self) -> None:
+        """Perform pending MQTT client unsubscribes."""
+        if not self._pending_unsubscribes:
+            return
+
+        topics = set(self._pending_unsubscribes)
+        self._pending_unsubscribes = set()
+
+        def _process_client_unsubscribes() -> tuple[int, int]:
+            """Initiate all subscriptions on the MQTT client and return the results."""
+            result, mid = self._mqttc.unsubscribe(list(topics))
+            return result, mid
+
+        async with self._paho_lock:
+            result, mid = await self.hass.async_add_executor_job(
+                _process_client_unsubscribes
+            )
+            _raise_on_error(result)
```

Comment:
Shouldn't this be in an if block below (again,mirroring https://github.com/home-assistant/core/pull/92172/files).

ie.
```
if result == 0:
  await self._wait_for_mid(mid)
else:
  _raise_on_error(result)
```